app/Http/Controllers/PostMarkController.php

```diff
@@ -69,4 +74,255 @@ public function webhook(Request $request)
+     *     ),
+     *     11 =>
+     *     array (
+     *       'Name' => 'X-Gm-Message-State',
+     *       'Value' => 'AOJu0Yxpbp1sRh17lNzg+pLnIx1jCn8ZFJQMgFuHK+6Z8RqFS5KKKTxR	8onpEbxWYYVUbrJFExNBHPD/3jdxqifCVVNaDmbpwHgmW5lHLJmA5vYRq5NFZ9OA6zKx/N6Gipr	iXE4fXmSqghFNTzy9V/RT08Zp+F5RiFh/Ta6ltQl8XfCPFfSawLz6cagUgt8bBuF4RqdrYmWwzj	ty86V5Br1htRNEFYivoXnNmaRcsD0tca1D23ny62O6RwWugrj1IpAYhViNyTZAWu+loKgfjJJoI	MsyiSU=',
+     *     ),
+     *     12 =>
+     *     array (
+     *       'Name' => 'X-Google-Smtp-Source',
+     *       'Value' => 'AGHT+IEdtZqbVI6j7WLeaSL3dABGSnWIXaSjbYqXvFvE2H+f2zsn0gknQ4OdTJecQRCabpypVF2ue91Jb7aKl6RiyEQ=',
+     *     ),
+     *     13 =>
+     *     array (
+     *       'Name' => 'X-Received',
+     *       'Value' => 'by 2002:a19:385a:0:b0:513:c876:c80a with SMTP id d26-20020a19385a000000b00513c876c80amr2586776lfj.34.1711282689140; Sun, 24 Mar 2024 05:18:09 -0700 (PDT)',
+     *     ),
+     *     14 =>
+     *     array (
+     *       'Name' => 'MIME-Version',
+     *       'Value' => '1.0',
+     *     ),
+     *     15 =>
+     *     array (
+     *       'Name' => 'References',
+     *       'Value' => '<CADfEuNsNFmNNCJDPjpS36amoLv2XEm41HmgYJT7Tj=R96PkxnA@mail.gmail.com>',
+     *     ),
+     *     16 =>
+     *     array (
+     *       'Name' => 'In-Reply-To',
+     *       'Value' => '<CADfEuNsNFmNNCJDPjpS36amoLv2XEm41HmgYJT7Tj=R96PkxnA@mail.gmail.com>',
+     *     ),
+     *     17 =>
+     *     array (
+     *       'Name' => 'Message-ID',
+     *       'Value' => '<CADfEuNvyCLsnp=CwJ3BF=-L6rn=o+DmUOPP6Cp4F-SO0p0hVwQ@mail.gmail.com>',
+     *     ),
+     *   ),
+     *   'Attachments' =>
+     *   array (
+     *      array (
+     *          'Content' => "base64-String",
+     *          'ContentLength' => 60164,
+     *          'Name' => 'Unbenannt.png',
+     *          'ContentType' => 'image/png',
+     *          'ContentID' => 'ii_luh2h8lg0',
+     *      )
+     *   ),
+     *  )
+     */
+    public function inboundWebhook(Request $request)
+    {
+
+        $input = $request->all();
+
+        if (!($request->has('token') && $request->get('token') == config('ninja.inbound_mailbox.inbound_webhook_token')))
+            return response()->json(['message' => 'Unauthorized'], 403);
+
+        if (!(array_key_exists("MessageStream", $input) && $input["MessageStream"] == "inbound") || !array_key_exists("To", $input) || !array_key_exists("From", $input) || !array_key_exists("MessageID", $input)) {
+            nlog('Failed: Message could not be parsed, because required parameters are missing.');
+            return response()->json(['message' => 'Failed. Missing/Invalid Parameters.'], 400);
+        }
+
+        $inboundEngine = new InboundMailEngine();
+
+        if ($inboundEngine->isInvalidOrBlocked($input["From"], $input["To"])) {
+            return response()->json(['message' => 'Blocked.'], 403);
+        }
+
+        $company = MultiDB::findAndSetDbByExpenseMailbox($input["To"]);
```

Comment:
Just to confirm on this one, this particular field would then also require to be unique in the system, ie company.expense_mailbox

Reply:
yes, thats right :)

Reply:
can this be resolved or are there any further questions on this topic?